amaranth/_utils.py

```diff
@@ -74,18 +74,39 @@ def decorator_like(*args, **kwargs):
                 return f(*args, **kwargs)
         return decorator_like
 
+_LINT_PATTERN = re.compile(r"^#\s*amaranth:\s*((?:\w+=\w+\s*)(?:,\s*\w+=\w+\s*)*)\n$")
+
 
 def get_linter_options(filename):
     first_line = linecache.getline(filename, 1)
     if first_line:
-        match = re.match(r"^#\s*amaranth:\s*((?:\w+=\w+\s*)(?:,\s*\w+=\w+\s*)*)\n$", first_line)
+        match = _LINT_PATTERN.match(first_line)
```

Comment:
Since we're using match groups I would rather keep the expression within the function. (The Python `re` module has an LRU cache for expressions, so under any normal circumstances it will be compiled only once anyway.)

Reply:
good to know! fixed